### PR TITLE
refactor: refine tencentcloud_security_group_lite_rule

### PR DESCRIPTION
**Pull request by Sherlock-Holo**
## refactor
rewrite the ingress and egress update logic

## doc
refine doc

### CODE REVIEW

**Comment by Sherlock-Holo** [comment]
> ```
> go test -v -run=TestAccTencentCloudSecurityGroupLiteRule -timeout 120m
> === RUN   TestAccTencentCloudSecurityGroupLiteRule_basic
> === PAUSE TestAccTencentCloudSecurityGroupLiteRule_basic
> === RUN   TestAccTencentCloudSecurityGroupLiteRule_update
> === PAUSE TestAccTencentCloudSecurityGroupLiteRule_update
> === CONT  TestAccTencentCloudSecurityGroupLiteRule_basic
> === CONT  TestAccTencentCloudSecurityGroupLiteRule_update
> --- PASS: TestAccTencentCloudSecurityGroupLiteRule_basic (2.68s)
> --- PASS: TestAccTencentCloudSecurityGroupLiteRule_update (8.50s)
> PASS
> ok      github.com/terraform-providers/terraform-provider-tencentcloud/tencentcloud     8.984s
> ```

File: tencentcloud/service_tencentcloud_vpc.go [hunk 1]
```diff
@@ -1397,74 +1405,205 @@ func (me *VpcService) DescribeSecurityGroups(ctx context.Context, sgId, sgName *
...
+	sgId string,
+	ingress, egress []VpcSecurityGroupLiteRule,
+	ingressModifyKind, egressModifyKind liteRuleModifyKind,
+) error {
 	logId := getLogId(ctx)
 
-	request := vpc.NewModifySecurityGroupPoliciesRequest()
-	request.SecurityGroupId = &sgId
-	request.SecurityGroupPolicySet = new(vpc.SecurityGroupPolicySet)
+	switch {
```
Comment (Himer) [inline tencentcloud/service_tencentcloud_vpc.go:1416]: 不用switch
用
if len(ingress ) set ingress 
if len(egress ) set egress

File: tencentcloud/service_tencentcloud_vpc.go [hunk 2]
```diff
@@ -1552,12 +1691,23 @@ func (me *VpcService) DescribeSecurityGroupPolices(ctx context.Context, sgId str
 func (me *VpcService) DetachAllLiteRulesFromSecurityGroup(ctx context.Context, sgId string) error {
 	logId := getLogId(ctx)
 
-	if err := me.modifyLiteRulesInSecurityGroup(ctx, sgId, nil, nil); err != nil {
-		log.Printf("[CRITAL]%s detach all lite rules from security group failed, reason: %v", logId, err)
-		return err
+	request := vpc.NewModifySecurityGroupPoliciesRequest()
+	request.SecurityGroupId = &sgId
+	request.SecurityGroupPolicySet = &vpc.SecurityGroupPolicySet{
+		Version: helper.String(strconv.Itoa(0)),
```
Comment (Himer) [inline tencentcloud/service_tencentcloud_vpc.go:1697]: helper.String("0")

File: tencentcloud/service_tencentcloud_vpc.go [hunk 3]
```diff
@@ -86,6 +86,14 @@ type VpcEniIP struct {
 	desc    *string
 }
 
+type liteRuleModifyKind int
+
+const (
+	noModifyLiteRule liteRuleModifyKind = iota + 1
+	updateLiteRule
+	deleteLiteRule
```
Comment (Himer) [inline tencentcloud/service_tencentcloud_vpc.go:94]: 常量加个vpc业务前缀

File: website/docs/r/security_group_lite_rule.html.markdown [hunk 4]
```diff
@@ -10,7 +10,7 @@ description: |-
 
 Provide a resource to create security group some lite rules quickly.
 
--> **NOTE:** It can't be used with tencentcloud_security_group_rule.
+-> **NOTE:** It can't be used with tencentcloud_security_group_rule, and don't create multi tencentcloud_security_group_rule resources, otherwise it may cause problems.
```
Comment (likexian) [inline website/docs/r/security_group_lite_rule.html.markdown:13]: multi  -> multiple

File: tencentcloud/service_tencentcloud_vpc.go [hunk 5]
```diff
@@ -1397,74 +1405,205 @@ func (me *VpcService) DescribeSecurityGroups(ctx context.Context, sgId, sgName *
...
+		})
 	}
 
-	if err := me.modifyLiteRulesInSecurityGroup(ctx, sgId, ingress, egress); err != nil {
+	// won't delete ingress and egress together in this method
+
+	return nil
+}
+
+func (me *VpcService) AttachLiteRulesToSecurityGroup(
```
Comment (oliverpei) [inline tencentcloud/service_tencentcloud_vpc.go:1598]: 这个函数和 modifyLiteRulesInSecurityGroup 一样，为什么不直接调用，还要再加一层调用

File: tencentcloud/service_tencentcloud_vpc.go [hunk 6]
```diff
@@ -1397,74 +1405,205 @@ func (me *VpcService) DescribeSecurityGroups(ctx context.Context, sgId, sgName *
...
+			request := vpc.NewModifySecurityGroupPoliciesRequest()
+			request.SecurityGroupId = &sgId
+			request.SecurityGroupPolicySet = new(vpc.SecurityGroupPolicySet)
+
+			for _, eg := range egress {
+				policy := &vpc.SecurityGroupPolicy{
+					Protocol:  helper.String(eg.protocol),
+					CidrBlock: helper.String(eg.cidrIp),
+					Action:    helper.String(eg.action),
+				}
```
Comment (oliverpei) [inline tencentcloud/service_tencentcloud_vpc.go:1473]: 这里可以简化下：
```
for i := range egress {
	policy := &vpc.SecurityGroupPolicy{
		Protocol:  &egress[i].protocol,
		CidrBlock: &egress[i].cidrIp,
		Action:   &egress[i].action,
	}
}
```
另外，SecurityGroupPolicy 的成员大写开头

**Pull request merged**